### ISSUE TITLE
Slack Notifications

### ISSUE THREAD

**Issue by sk-sahu** [seqeralabs/nf-tower#222]
This would be great if the inbuilt notification feature of Nextflow (which I can see already added to the nf-tower, I'm receiving emails.) can be integrated with [Slack](https://slack.com/).

**Comment by pditommaso** [comment]
Could you manage to find details about Slack API and auth?

**Comment by sk-sahu** [comment]
Sure, I'm already looking into it.

**Comment by sk-sahu** [comment]
Till now what I tried to figure out an [Events API](https://api.slack.com/events-api) should be able to help here. Its supports [OAuth permission](https://api.slack.com/docs/oauth-scopes) also.

Need to create a slack app for this, which can be the endpoint to trigger an Event.

I have no much idea about it. But I'm reading. I'll try to help whatever way I can.

**Comment by pditommaso** [comment]
Not sure. I think that allows you to receive in your app Slack notification. Instead this issue aims the opposite I guess. Maybe [this](https://api.slack.com/methods/chat.postEphemeral) can help.

**Comment by sk-sahu** [comment]
Yes, this makes more sense.  Just with an Incoming Webhooks, a notification can be pushed to a given channel.

**Comment by sk-sahu** [comment]
<a href="https://imgbb.com/"><img src="https://i.ibb.co/m61kkTQ/nf-notify.png" alt="nf-notify" border="0"></a><br /><a target='_blank' href='https://clueyblog.com/which-hairstyle-suits-me'></a><br />

**Comment by pditommaso** [comment]
Nice

**Comment by ewels** [comment]
We are super excited about this feature for @nf-core - it would be amazing to have the full scale test runs post their progress into the relevant pipeline's Slack channel automatically...

**Comment by pditommaso** [comment]
Me too! we need some slack gurus that hacks into their API.

**Comment by ewels** [comment]
I've played with it a little to pull out stuff like the [number of Slack users](https://nf-co.re/stats#slack). I found it quite easy to work with - the docs are generally pretty good. The "Tester" tab for each method is super useful especially.

I guess you'll need to set up a new [Slack App](https://api.slack.com/apps) for each running instance of Tower, then you can probably put these details into a config file. The tricky bit is then figuring out what approach to use - you can do this in a [lot of different ways](https://api.slack.com/start/planning/choosing). The easiest is probably by using the [web API](https://api.slack.com/web) I guess. Then at some point you can submit this to the [Slack App Directory](https://api.slack.com/reference/slack-apps/directory-submission-checklist) so that people can find it and add it to their Slack workspace.

My suggestion would be to build an app with some [slash commands](https://api.slack.com/interactivity/slash-commands) for subscribing and unsubscribing the current channel to tower. You could then give arguments to filter for specific workflows. For example `/tower subscribe` could subscribe to _all_ workflow events in tower.nf. Then `/tower subscribe workflow:nf-core/rnaseq` could subscribe to just that pipeline. And so on.. Would need some thought about all of the different methods you want and how to structure them in the slack slash command syntax.

The [GitHub Slack integration](https://github.com/integrations/slack/) repo is probably a good reference. It's pretty complex, but you can see at least how the slack commands are structured with different actions and filters.

**Comment by pditommaso** [comment]
... `/tower launch ..` 🤯
